dags/iop/parser.py

```diff
@@ -100,3 +105,80 @@ def _get_extracted_arxiv_eprint_value(self, article):
             self.logger.error("The arXiv value is not valid.", dois=self.dois)
         except AttributeError:
             self.logger.error("No arXiv eprints found", dois=self.dois)
+
+    def _get_authors(self, article):
+        contrib_types = article.findall(
+            "front/article-meta/contrib-group/contrib[@contrib-type='author']"
+        )
+        authors = []
+        surname = ""
+        given_names = ""
+
+        for contrib_type in contrib_types:
+            try:
+                surname = contrib_type.find("name/surname").text
+            except AttributeError:
+                self.logger.error("Surname is not found in XML", dois=self.dois)
+            try:
+                given_names = contrib_type.find("name/given-names").text
+            except AttributeError:
+                self.logger.error("Given_names is not found in XML", dois=self.dois)
+
+            reffered_ids = contrib_type.findall("xref[@ref-type='aff']")
+            affiliations = [
+                self._get_affiliation_value(article, reffered_id)
+                for reffered_id in reffered_ids
+                if self._get_affiliation_value(article, reffered_id) is not None
+            ]
+
+            if "collaboration" not in given_names.lower():
+                author = clean_dict(
+                    {
+                        "surname": surname,
+                        "given_names": given_names,
+                        "affiliations": affiliations,
+                    }
```

Comment:
Are you aware that if there is an `AttributeError` error above and it won't be the first author from extracted authors, it will get the previous extracted name/given_names?

Reply:
No, because it has to come from the same contrib_type, I will write a test for it

Reply:
I don't get it then. Let's say you have `contrib_types` with two entries. You iterate through this iterable. In the first iteration, you get an author, `surname=Test`, `given_names='Author`. You go to the second iteration, `surname` is not found for some reason, but from the previous iteration the value of the `surname` variable is `Test`. So you end up putting the name of the first author in the second author, am I missing something?

Reply:
I will try to explain. Let's take the example you gave me:
contrib_types has to entries, the first entry has everything, so the dict will look like this:
``{     "surname": 'Test',
            "given_names": 'Author',
            "affiliations": affiliations,
    }``

The second one doesn’t have surname, so it won't be in the dict will be without this value:
``{     
            "given_names": 'Name of the second one',
            "affiliations": affiliations,
    }``

Is it more clear?

Reply:
test: https://github.com/cern-sis/workflows/blob/3adb98ad19e895cb7efa61696d97cda7e3d7ad67/tests/units/iop/test_iop_parser.py#L788-L832